traitsui/tree_node_renderer.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2019, Enthought Inc.
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+from abc import abstractmethod
+
+from traits.api import ABCHasStrictTraits, Bool
+
+
+class AbstractTreeNodeRenderer(ABCHasStrictTraits):
+    """ Abstract base class for renderers of tree node items.
+
+    This is currently only supported for Qt.
+    """
+
+    #: Whether the renderer handles rendering everything
+    handles_all = Bool(False)
+
+    #: Whether the renderer handles rendering any text
+    handles_text = Bool(True)
+
+    #: Whether the renderer handles rendering the icon or other decoration
+    handles_icon = Bool(False)
+
+    @abstractmethod
+    def paint(self, node, column, object, paint_context):
+        """ Render the node.
+
+        Parameters
+        ----------
+        node : ITreeNode instance
+            The tree node to render.
+        column : int
+            The column in the tree that should be rendererd.
+        object : object
+            The underlying object being edited.
+        paint_context : object
+            A toolkit-dependent context for performing paint operations.
+        """
```

Comment:
You should probably mention the optional return value here

Reply:
Yep, although I think I have to get rid of it and replace it with an explicit `size_hint` method to fix the Python 3 issue cleanly.